### PR TITLE
Adds ability to control block and stack debug output via CMake

### PR DESCRIPTION
**Pull request by dotjrich**
Also updates README to detail how to enable various debug options.

A typo related to running tests has also been fixed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,18 @@ cmake_minimum_required(VERSION 3.1)
...
+option(USE_BLOCK_DEBUG "Enable block debugging" OFF)
+option(USE_STACK_DEBUG "Enable stack debugging" OFF)
+
+# Turn debug defs on if they're enabled.
+if (USE_BLOCK_DEBUG)
+  add_definitions(-DBLOCK_DEBUG)
+endif()
+if (USE_STACK_DEBUG)
+  add_definitions(-DSTACK_DEBUG)
+endif()
```
Comment (zrax) [inline CMakeLists.txt:17]: 4 space indentation please

File: CMakeLists.txt [hunk 2]
```diff
@@ -4,6 +4,18 @@ cmake_minimum_required(VERSION 3.1)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+# Debug options.
+option(USE_BLOCK_DEBUG "Enable block debugging" OFF)
+option(USE_STACK_DEBUG "Enable stack debugging" OFF)
```
[the remark this answers is not in the document]
Comment (dotjrich) [reply CMakeLists.txt]: Fixed!

File: CMakeLists.txt (hunk 1 above)
Comment (dotjrich) [reply CMakeLists.txt]: Fixed!

**Review by zrax — APPROVED** [review]

**Comment by zrax** [comment]
Thanks for the contribution!

**Pull request merged**